src/libcollections/vec_deque.rs

```diff
@@ -2158,6 +2158,18 @@ impl<A: PartialEq> PartialEq for VecDeque<A> {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<A: Eq> Eq for VecDeque<A> {}
 
+#[stable(feature = "vec-deque-partial-eq-slice", since = "")]
+impl<'a, A: PartialEq> PartialEq<&'a [A]> for VecDeque<A> {
```

Comment:
I think `PartialEq<[A]>` should be the primary one. Does that change the utility of it? The type inference problems? (I guess it's the same.)